brainiak/utils/fmrisim_real_time_generator.py

```diff
@@ -287,41 +288,55 @@ def _get_input_defaults(data_dict):
     """
 
     # Load in the ROIs
-    if data_dict['ROI_A_file'] is None:
+    if data_dict.get('ROI_A_file') is None:
         vol = resource_stream(__name__, "sim_parameters/ROI_A.nii.gz").read()
         ROI_A_file = Nifti1Image.from_bytes(gzip.decompress(vol)).get_data()
     else:
         ROI_A_file = data_dict['ROI_A_file']
 
-    if data_dict['ROI_B_file'] is None:
+    if data_dict.get('ROI_B_file') is None:
         vol = resource_stream(__name__, "sim_parameters/ROI_B.nii.gz").read()
         ROI_B_file = Nifti1Image.from_bytes(gzip.decompress(vol)).get_data()
     else:
         ROI_B_file = data_dict['ROI_B_file']
 
     # Get the path to the template
-    if data_dict['template_path'] is None:
+    if data_dict.get('template_path') is None:
         vol = resource_stream(__name__,
                               "sim_parameters/sub_template.nii.gz").read()
         template_path = Nifti1Image.from_bytes(gzip.decompress(vol)).get_data()
     else:
         template_path = data_dict['template_path']
 
     # Load in the noise dict if supplied
-    if data_dict['noise_dict_file'] is None:
+    if data_dict.get('noise_dict_file') is None:
         file = resource_stream(__name__,
                                'sim_parameters/sub_noise_dict.txt').read()
         noise_dict_file = file
     else:
         noise_dict_file = data_dict['noise_dict_file']
 
     # Update defaults if they are missing
+    if 'numTRs' not in data_dict:
+        data_dict['numTRs'] = 200
     if 'trDuration' not in data_dict:
         data_dict['trDuration'] = 2
     if 'isi' not in data_dict:
         data_dict['isi'] = 6
     if 'burn_in' not in data_dict:
         data_dict['burn_in'] = 6
+    if 'event_duration' not in data_dict:
+        data_dict['event_duration'] = 10
+    if 'scale_percentage' not in data_dict:
+        data_dict['scale_percentage'] = 0.5
+    if 'multivariate_pattern' not in data_dict:
+        data_dict['multivariate_pattern'] = False
+    if 'different_ROIs' not in data_dict:
+        data_dict['different_ROIs'] = False
+    if 'save_dicom' not in data_dict:
+        data_dict['save_dicom'] = False
+    if 'save_realtime' not in data_dict:
+        data_dict['save_realtime'] = False
```

Comment:
I was thinking that we could build the dictionary the other way around: start with a default `dict` and `update` it with the user data in `data_dict`. It would be easier to maintain and no trigger the C901 complexity error.

Reply:
I see. One thing to consider is that the generate_data() function can be called from main(), in which case it will add the args params to the data_dict, but it can also be called directly from another python script (how I use it in the rt-cloud sample) in which case the passed in data_dict will just contain the settings they want to modify. So I guess you are suggesting that generate_data() start with a default_dict and then merge whatever dict is passed into it with the default_dict, is that right?

Reply:
That's right.